packages/@uppy/core/src/Uppy.js

```diff
@@ -362,29 +363,34 @@ class Uppy {
 
   /*
   * @constructs
-  * @param { Error } error
+  * @param { Error[] } errors
   * @param { undefined } file
   */
   /*
   * @constructs
   * @param { RestrictionError } error
-  * @param { UppyFile | undefined } file
   */
-  #informAndEmit (error, file) {
-    const { message, details = '' } = error
+  #informAndEmit (errors) {
+    for (const error of errors) {
+      const { file, isRestriction } = error
 
-    if (error.isRestriction) {
-      this.emit('restriction-failed', file, error)
-    } else {
-      this.emit('error', error)
+      if (isRestriction) {
+        this.emit('restriction-failed', file, error)
+      } else {
+        this.emit('error', error)
+      }
+      this.log(error, 'warning')
     }
-    this.info({ message, details }, 'error', this.opts.infoTimeout)
-    this.log(error, 'warning')
+
+    // don't flood the user: only show the first 5 toasts
+    errors.filter((error) => error.isUserFacing).slice(0, 5).forEach(({ message, details = '' }) => {
```

Comment:
Only the first 5 toasts? Wouldn't this be confused too? What about aggregating all into a single inform?

Reply:
Yes, do you have any suggestion for how to aggregating them? You mean just say something like `5 files did not pass restrictions`?

Reply:
the problem is that there are different file names for different restrictions so it's not that easy. i refactored now so that we show up to 4 restriction errors. then if there are more than 4, it will show an additrional toast: `%{count} additional restrictions were not fulfilled` where count is restrictionErrors.length - 4